sympy/polys/rootconditions.py

```diff
@@ -0,0 +1,264 @@
+        return _routh_hurwitz_qq(f, K)
+
+    elif K.is_ZZ or K.is_RR:
+        pq = dup_convert(f, K, QQ)
+        conds = _routh_hurwitz_qq(pq, QQ)
+        return dup_convert(conds, QQ, K)
+
+    elif K.is_PolynomialRing:
+        return _routh_hurwitz_poly(f, K)
+
+    elif K.is_FractionField:
+        _, pp = dup_clear_denoms(f, K, convert=True)
+        conds = _routh_hurwitz_poly(pp, K)
+        return dup_convert(conds, K.get_ring(), K)
+
+    else:
+        return _routh_hurwitz_exraw(dup_convert(f, K, EXRAW))
+
+
+def _routh_hurwitz_qq(p: list[MPQ], K: RationalField) -> list[MPQ]:
+    if len(p) < 2:
+        return [K.one]
+    elif p[0] * p[1] <= 0:
+        return [-K.one]
+    elif len(p) == 2:
+        return [K.one]
+
+    qs = p.copy()
+    for i in range(1, len(p), 2):
+        qs[i - 1] -= p[i] * p[0] / p[1]
+    qs = qs[1:]
+
+    return _routh_hurwitz_qq(qs, K)
+
+
+def _routh_hurwitz_poly(p: list[PolyElement[Er]], K: PolynomialRing[Er]):
+    return _rec_routh_hurwitz_poly(p, [], K)
+
+
+def _rec_routh_hurwitz_poly(p: list[PolyElement[Er]],
+                            previous_cond: list[PolyElement[Er]],
+                            K: PolynomialRing[Er]):
+    if len(p) < 2:
+        return [K.one]
+
+    if len(p) == 2:
+        return previous_cond + [_clear_cond_poly(p[0] * p[1], previous_cond, K)]
+
+    qs = [p[1] * qi for qi in p]
+    for i in range(1, len(p), 2):
+        qs[i - 1] -= p[i] * p[0]
+    qs = qs[1:]
+
+    cond = p[0] * p[1]
+
+    # This check is really important. It stops the recursion immediately and
+    # avoid a possible infinite while loop in _clear_cond_poly.
+    if cond.is_zero:
+        return [K.zero]
+
+    cond = _clear_cond_poly(cond, previous_cond, K)
+
+    # These checks are also important, because adding 1 or -1 to the previous
+    # conditions will lead to an infinite loop in the next _clear_cond_poly.
+    if cond == K(-1):
+        return [K(-1)]
+
+    if not cond.is_one:
+        previous_cond.append(cond)
+
+    return _rec_routh_hurwitz_poly(qs, previous_cond, K)
+
+def _clear_cond_poly(cond: PolyElement[Er], previous_cond, K):
+    # Divide out factors known to be positive from previous conditions.
+
+    # There are not controls in that functions on cond and previous_cond,
+    # we assume that at this point, there are no zeroes, ones and negative ones
+    # in previous_cond.
+    # If there are, there will be an infinite while loop.
+    for c in previous_cond:
+        cond_quo, r = K.div(cond, c)
+        while not r:
+            cond = cond_quo
+            cond_quo, r = K.div(cond, c)
+
+    # Remove factors of even degree and reduce odd degree factors
+    common_factor, facs_m = cond.sqf_list()
+
+    cond = K(prod([fac for fac, m in facs_m if m % 2 == 1]))
+    if common_factor < 0:
+        cond = cond * K(-1)
+
+    return cond
+
+
+def _routh_hurwitz_exraw(p: list[Er]) -> list[Er]:
+    if all(c.is_number for c in p):
+        return _calc_conditions_div(p)
```

Comment:
`is_number` is not the right check here. It should probably be `is_Number` i.e. you only want this if the coefficients are all Rational or Float but not e.g. something like `sqrt(2)`.